File: Go/Numbers/pi.go
package main

import (
	"fmt"
	"log"
	"math"
	"os"
	"strconv"
)

const (
	MAX = 25
)

func main() {
	programArgs := os.Args

	if argsLen := len(programArgs); argsLen == 1 || argsLen > 2 {
		log.Fatal("Wrong number of arguments!")
		os.Exit(1)
	}

	howMany, err := strconv.ParseInt(programArgs[1], 10, 64)
	if err != nil {
		log.Fatal(err)
		os.Exit(2)
	}

	if howMany > MAX {
		log.Fatalf("Max is %v.\n", MAX)
		os.Exit(3)
	}

	var format string = "%." + strconv.Itoa(int(howMany)) + "f"
	fmt.Printf("Pi number with %d decimal numbers "+format+"\n", howMany, math.Pi)
}

File: JAVA/Numbers/pi.java
import java.lang.Math;

class Pi {
    public static final int MAX = 25;

    public static void main (String[] args) {
        if (args.length == 0 || args.length > 1) {
            System.out.println("Wrong number of arguments!");
            System.exit(-1);
        }

        int howMany = Integer.parseInt(args[0]);
        if (howMany > Pi.MAX) {
            System.out.println("You can request for up to " + Pi.MAX + " numbers.");
            System.exit(-2);
        }

        String format = "%." + howMany + "f";
        System.out.println(String.format("Pi up to " + howMany + " numbers: " + format, Math.PI));
    }
}
